### PR TITLE
Refactor rides to be only accessible with getters

### PR DESCRIPTION
**Pull request by janisozaur**
This changes how rides are accessed from macros to getter functions.

### CODE REVIEW

File: src/peep/peep.c [hunk 1]
```diff
@@ -5645,21 +5645,21 @@ void get_arguments_from_action(rct_peep* peep, uint32 *argument_1, uint32* argum
 	case PEEP_STATE_LEAVING_RIDE:
 	case PEEP_STATE_ENTERING_RIDE:
 		*argument_1 = STR_ON_RIDE;
-		ride = g_ride_list[peep->current_ride];
+		ride = *get_ride(peep->current_ride);
```
Comment (duncanspumpkin) [inline src/peep/peep.c]: change this so that we dont copy the ride make ride a pointer.

File: src/ride/ride.c [hunk 2]
```diff
@@ -156,9 +156,29 @@ static void ride_update_vehicle_colours(int rideIndex);
...
+	{
+		log_error("invalid index %d for ride", index);
+		return NULL;
+	}
+	return &g_ride_list[index];
+}
+
+rct_ride_type *get_ride_entry(int index)
+{
+	return gRideTypeList[index];
```
Comment (duncanspumpkin) [inline src/ride/ride.c]: There are max and min ride entries plus you can check for Null pointers

File: src/ride/ride.c [hunk 3]
```diff
@@ -156,9 +156,29 @@ static void ride_update_vehicle_colours(int rideIndex);
...
+
+rct_ride_measurement *get_ride_measurement(int index)
+{
+	return &(RCT2_ADDRESS(RCT2_ADDRESS_RIDE_MEASUREMENTS, rct_ride_measurement)[index]);
+}
+
 rct_ride_type *ride_get_entry(rct_ride *ride)
 {
-	return GET_RIDE_ENTRY(ride->subtype);
+	return gRideTypeList[ride->subtype];
```
Comment (duncanspumpkin) [inline src/ride/ride.c]: You may as well just make this call the other function then you don't need to copy any max/min checks.

File: src/peep/peep.c (hunk 1 above)
Comment (IntelOrca) [inline src/peep/peep.c]: I can't believe it was copying the ride in this function, that's insane!

**Comment by IntelOrca** [comment]
Could you also change g_ride_list to gRideList?

File: src/peep/peep.c (hunk 1 above)
Comment (duncanspumpkin) [inline src/peep/peep.c]: It probably gets optimised out anyway.

File: src/ride/ride.c (hunk 2 above)
Comment (janisozaur) [inline src/ride/ride.c]: For rides there are, but I haven't seen any for `rct_ride_type`, only hint is https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/ride/ride.h#L440-L533

**Pull request merged**